c7n/resources/vpc.py

```diff
@@ -1600,3 +1600,79 @@ class resource_type(object):
         name = 'KeyName'
         date = None
         dimension = None
+
+
+@Vpc.action_registry.register('set-vpc-flow')
+@Subnet.action_registry.register('set-vpc-flow')
+@NetworkInterface.action_registry.register('set-vpc-flow')
+class CreateFlowLogs(BaseAction):
+    """Create flow logs for a network resource
+
+    :example:
+
+    .. code-block: yaml
+
+        policies:
+          - name: vpc-enable-flow-logs
+            resource: vpc
+            filters:
+              - type: flow-logs
+                enabled: false
+            actions:
+              - type: set-vpc-flow
+                DeliverLogsPermissionArn: arn:iam:role
+                LogGroupName: /custodian/vpc/flowlogs/
+    """
+    permissions = ('ec2:CreateFlowLogs',)
+    schema = {
+        'type': 'object',
+        'additionalProperties': False,
+        'properties': {
+            'type': {'enum': ['set-vpc-flow']},
+            'DeliverLogsPermissionArn': {'type': 'string'},
+            'LogGroupName': {'type': 'string'},
+            'TrafficType': {'type': 'string',
+                            'enum': ['ACCEPT', 'REJECT', 'ALL']},
+            'required': ['LogGroupName']
+        }
+    }
+
+    def validate(self):
+        if not self.data.get('DeliverLogsPermissionArn'):
+            if self.data.get('LogGroupName'):
+                raise ValueError('DeliverLogsPermissionArn required when '
+                                 'specifying LogGroupName')
+        return self
+
+    def delete_flow_logs(self, client, rids):
+        flow_logs = client.describe_flow_logs(
+            Filters=[{'Name': 'resource-id', 'Values': rids}])['FlowLogs']
+        client.delete_flow_logs(FlowLogIds=[f['FlowLogId'] for f in flow_logs])
```

Comment:
we'd want to catch on the exception condition that the flow log isn't present and pass.